ui/app/routes/evaluations/LaunchEvaluationModal.tsx

```diff
@@ -127,22 +127,121 @@ function DatasetSelector({
   );
 }
 
-function EvaluationForm({ dataset_names }: { dataset_names: string[] }) {
+interface EvaluationsFormValues {
+  dataset_name: string | null;
+  evaluation_name: string | null;
+  variant_name: string | null;
+  concurrency_limit: string;
+  inference_cache: InferenceCacheSetting;
+}
+
+interface EvaluationsFormState extends EvaluationsFormValues {
+  advancedParamsOpen: boolean;
+  _lastAction: "sync" | "user" | null;
+}
+
+type FieldUpdateAction<K extends keyof EvaluationsFormValues> = {
+  type: "UPDATE_FIELD";
+  fieldName: K;
+  value: EvaluationsFormValues[K];
+};
+
+type FormUpdateAction =
+  | {
+      [K in keyof EvaluationsFormValues]: FieldUpdateAction<K>;
+    }[keyof EvaluationsFormValues]
+  | { type: "SYNC_FORM_STATE"; state: Partial<EvaluationsFormState> }
+  | { type: "TOGGLE_ADVANCED_PARAMS"; open?: boolean }
+  | { type: "USER_ACTION" };
+
+function EvaluationForm({ datasetNames }: { datasetNames: string[] }) {
   const fetcher = useFetcher();
   const config = useConfig();
   const evaluation_names = Object.keys(config.evaluations);
-  const [selectedEvaluationName, setSelectedEvaluationName] = useState<
-    string | null
-  >(null);
-  const [selectedDatasetName, setSelectedDatasetName] = useState<string | null>(
-    null,
-  );
-  const [selectedVariantName, setSelectedVariantName] = useState<string | null>(
-    null,
+
+  const [formState, formUpdate] = useReducer(
```

Comment:
Figured using a reduce here consolidates a bunch of logic and avoids having React batch a bunch of state updates every time we seed the form with new state for several fields.